packages/next-swc/crates/napi/src/parse.rs

```diff
@@ -0,0 +1,76 @@
+use crate::{
+  get_compiler,
+  util::{deserialize_json, CtxtExt, MapErr},
+};
+use anyhow::Context as _;
+use napi::{CallContext, Either, Env, JsObject, JsString, JsUndefined, Task};
+use std::{sync::Arc};
+use swc::{try_with_handler, config::ParseOptions, Compiler};
+use swc_common::FileName;
+use swc_ecmascript::ast::Program;
+
+pub struct ParseTask {
+  pub c: Arc<Compiler>,
+  pub filename: FileName,
+  pub src: String,
+  pub options: String,
+}
+
+pub fn complete_parse<'a>(env: &Env, program: Program, _c: &Compiler) -> napi::Result<JsString> {
+  let s = serde_json::to_string(&program)
+      .context("failed to serialize Program")
+      .convert_err()?;
+  env.create_string_from_std(s)
+}
+
+impl Task for ParseTask {
+  type Output = Program;
+  type JsValue = JsString;
+
+  fn compute(&mut self) -> napi::Result<Self::Output> {
+      let options: ParseOptions = deserialize_json(&self.options).convert_err()?;
+      let fm = self
+          .c
+          .cm
+          .new_source_file(self.filename.clone(), self.src.clone());
+      let program = try_with_handler(self.c.cm.clone(), false, |handler| {
+          self.c.parse_js(
+              fm,
+              &handler,
+              options.target,
+              options.syntax,
+              options.is_module,
+              options.comments,
+          )
+      })
+      .convert_err()?;
+
+      Ok(program)
+  }
+
+  fn resolve(self, env: Env, result: Self::Output) -> napi::Result<Self::JsValue> {
+      complete_parse(&env, result, &self.c)
+  }
+}
+
+#[js_function(3)]
+pub fn parse(ctx: CallContext) -> napi::Result<JsObject> {
+    let c = get_compiler(&ctx);
+    let src = ctx.get_deserialized(0)?;
```

Comment:
This should be changed to use `JsString`, if we are changing input type to `string` (from `Buffer`)

Reply:
I'm getting this error if I change it to JsString, does it mean napi-rs requires it to be String?
```
  --> crates/napi/src/parse.rs:26:6                                                               │
   |                                                                                              │
26 | impl Task for ParseTask {                                                                    │
   |      ^^^^ `*mut napi_env__` cannot be sent between threads safely                            │
   |                                                                                              │
   = help: within `ParseTask`, the trait `Send` is not implemented for `*mut napi_env__`          │
   = note: required because it appears within the type `napi::js_values::value::Value`            │
   = note: required because it appears within the type `JsString`
```

Reply:
Yes, it should be converted to `String` from the `parse` function.

https://github.com/swc-project/swc/blob/7bccbcc880d29d0379886bf59ab93fc0ed308517/crates/node/src/parse.rs#L104

```rust
  let src = ctx.get::<JsString>(0)?.into_utf8()?.as_str()?.to_string();
```